src/library/scanner/libraryscanner.cpp

```diff
@@ -145,8 +142,11 @@ void LibraryScanner::run() {
             kLogger.info() << "Cleaning up database...";
             PerformanceTimer timer;
             timer.start();
-            auto numRows = execCleanupQuery(dbConnection,
-                    kDeleteOrphanedLibraryScannerDirectories);
+            const auto sqlCmd =
+                    QString("delete from LibraryHashes where hash <> %1 "
+                            "and directory_path not in (select directory from track_locations)")
+                            .arg(mixxx::invalidCacheKey());
```

Comment:
Even if that does not apply here, inserting parameters manually into an SQL query reeks of SQL injections. I'd strongly prefer to use prepare/bind/exec. 

Also, can you use upper case for the SQL keywords, i. e. `DELETE FROM Library Hashes WHERE...`? This make it more readable IMHO.

Reply:
I agree. This once was a hard-coded command without any parameters and the "invalid" hash value encoded implicitly.